src/ol/renderer/layerrenderer.js

```diff
@@ -151,7 +151,7 @@ ol.renderer.Layer.prototype.loadImage = function(image) {
     // listener (a noop if the listener was already registered)
     goog.asserts.assert(imageState == ol.ImageState.IDLE ||
         imageState == ol.ImageState.LOADING);
-    goog.events.listenOnce(image, goog.events.EventType.CHANGE,
+    goog.events.listen(image, goog.events.EventType.CHANGE,
```

Comment:
But registerDisposable only associates the image with the source so that the image gets disposed when the source gets disposed? Though I am not sure when or if the source will be disposed.
I'll investigate a bit more to confirm the GC cleans up.